sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -327,6 +338,10 @@ func (n *ProcessSizedElementsAndRestrictions) ProcessElement(_ context.Context,
 		}
 	}
 
+	if n.cweInv != nil {
+		n.PDo.we = n.cweInv.Invoke()
```

Comment:
Creating a watermark estimator doesn't need to be done for each element, does it? If not, this should go in StartBundle.

Reply:
It actually does - watermark estimation is done at the element/restriction level. This will become more important when we're taking in watermark state which is created while giving the user element/restriction access

Reply:
For Daniel's benefit: In the contexts where Watermark estimation is being used, often only "1" element will be processed, since it's an extremely high fanout from that 1 element. So Bad for micro benchmarking, but probably fine in practice.

Ultimately, this feeds into my desire to have a single Very Good example of how to write a Streaming SDF with all the bells and whistles, so folks don't end up hurting themselves with all the new features we're adding.